src/dataclass_binder/_impl.py

```diff
@@ -314,6 +325,15 @@ def _bind_to_single_type(self, value: object, field_type: type, context: str) ->
                 if not isinstance(value, str):
                     raise TypeError(f"Expected TOML string for path '{context}', got '{type(value).__name__}'")
                 return field_type(value)
+            elif issubclass(field_type, ReprEnum):
+                return field_type(value)
+            elif issubclass(field_type, Enum):
+                if not isinstance(value, str):
+                    raise TypeError(f"'{value}' is not a valid key for enum '{field_type}', must be of type str")
+                for enum_value in field_type:
+                    if enum_value.name.lower() == value.lower():
+                        return enum_value
+                raise TypeError(f"'{value}' is not a valid key for enum '{field_type}', could not be found")
```

Comment:
This message should also include `{context}`, to tell the end user which exact TOML field contains the problem.